npm/src/controller/oauth.ts

```diff
@@ -430,10 +481,20 @@ export class OAuthController implements IOAuthController {
       codeVal.session = session;
     }
 
-    await this.codeStore.put(code, codeVal);
-
-    if (session && session.redirect_uri && !allowed.redirect(session.redirect_uri, samlConfig.redirectUrl)) {
-      throw new JacksonError('Redirect URL is not allowed.', 403);
+    try {
+      await this.codeStore.put(code, codeVal);
+    } catch (err: unknown) {
+      if (!isIdPFlow && session?.redirect_uri) {
```

Comment:
We should standardize this for both flows by using `(session && session.redirect_uri) || samlConfig.defaultRedirectUrl` to figure out the redirect_uri to take the user to